core/src/main/java/org/kiwix/kiwixmobile/core/main/CompatFindActionModeCallback.kt

```diff
@@ -0,0 +1,213 @@
+  private val tag = "CompatFindActionMode"
+  @JvmField var isActive: Boolean
+
+  @SuppressLint("InflateParams")
+  private val customView: View = LayoutInflater.from(context).inflate(R.layout.webview_search, null)
+  private val editText: EditText
+  private val findResultsTextView: TextView
+  private var webView: WebView? = null
+  private val input: InputMethodManager
+  private var actionMode: ActionMode? = null
+
+  init {
+    editText = customView.findViewById(R.id.edit)
+    editText.setOnClickListener(this)
+    findResultsTextView = customView.findViewById(R.id.find_results)
+    input = context.getSystemService(Context.INPUT_METHOD_SERVICE) as InputMethodManager
+    isActive = false
+    setText("")
+  }
+
+  fun setActive() {
+    isActive = true
+  }
+
+  fun finish() {
+    actionMode?.finish()
+    webView?.clearMatches()
+  }
+
+  // Place text in the text field so it can be searched for.  Need to press
+  // the find next or find previous button to find all of the matches.
+  fun setText(text: String?) {
+    editText.setText(text)
+    val span: Spannable = editText.text
+    val length = span.length
+
+    // Ideally, we would like to set the selection to the whole field,
+    // but this brings up the Text selection CAB, which dismisses this
+    // one.
+    Selection.setSelection(span, length, length)
+
+    // Necessary each time we set the text, so that this will watch
+    // changes to it.
+    span.setSpan(this, 0, length, Spannable.SPAN_INCLUSIVE_INCLUSIVE)
+  }
+
+  // Set the WebView to search.  Must be non null, and set before calling startActionMode.
+  fun setWebView(webView: WebView?) {
+    if (null == webView) {
+      throw AssertionError(
+        "WebView supplied to CompatFindActionModeCallback cannot be null"
+      )
+    }
+    this.webView = webView
+    findResultsTextView.visibility = View.VISIBLE
+    this.webView?.setFindListener { activeMatchOrdinal: Int, numberOfMatches: Int, _: Boolean ->
+      val result: String = when {
+        editText.text.toString().isEmpty() -> {
+          ""
+        }
+        numberOfMatches == 0 -> {
+          "0/0"
+        }
+        else -> {
+          (activeMatchOrdinal + 1).toString() + "/" + numberOfMatches
+        }
+      }
+      findResultsTextView.text = result
+    }
+  }
+
+  // Move the highlight to the next match.
+  // If true, find the next match further down in the document.
+  // If false, find the previous match, up in the document.
+  private fun findNext(next: Boolean) {
+    if (webView == null) {
+      throw AssertionError("No WebView for CompatFindActionModeCallback::findNext")
+    }
+    webView?.findNext(next)
+  }
+
+  // Highlight all the instances of the string from mEditText in mWebView.
+  fun findAll() {
+    if (webView == null) {
+      throw AssertionError("No WebView for CompatFindActionModeCallback::findAll")
+    }
+    val find: CharSequence? = editText.text
+    if (find == null || find.isEmpty()) {
```

Comment:
replaceable with `find?.isEmpty() == true`

Reply:
also `find` is a strange name, `searchTerm`/`textToFind`? I dunno

Reply:
I'll rename `textToFind`. But `find?.isEmpty() == true` will evaluate to false when `find = null`

Reply:
We can flip the blocks and use isNotEmpty

Reply:
yeah, my mistake, that sounds good

Reply:
7770d6a